src/cloudant/document.py

```diff
@@ -351,5 +401,5 @@ def put_attachment(self, attachment, content_type, data, headers=None):
             headers=headers
         )
         resp.raise_for_status()
-
+        self.fetch()
```

Comment:
I definitely don't see this as a sidetrack.  This is actually the feedback I was looking for w.r.t. the double fetch in the attachment management.  

I just checked and unfortunately it does not appear that the response headers contain what the docs promise and the content length looks to differ by 1 when I ran my test.  In fact, the example provided with the link you referenced from the CouchDB docs shows no Content-MD5 entry in the response header and the content-length differing by 1 as well.  I suppose a little more digging is needed here...  To that end, I've opened https://cloudant.fogbugz.com/f/cases/55930 to track optimization of the library and I've referenced this conversation there.  I'll make the necessary changes to the `document.delete_attachment` method and we can circle back to the `document.put_attachment` optimzation at a later time.